back/engines/commercial/report_builder/app/services/report_builder/queries/visitors_languages.rb

```diff
@@ -0,0 +1,57 @@
+module ReportBuilder
+  class Queries::VisitorsLanguages < ReportBuilder::Queries::Base
+    def run_query(
+      start_at: nil,
+      end_at: nil,
+      project_id: nil,
+      exclude_roles: nil,
+      **_other_props
+    )
+      start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
+
+      pageviews = ImpactTracking::Pageview
+        .where(created_at: start_date..end_date)
+
+      if project_id.present?
+        pageviews = pageviews.where(project_id: project_id)
+      end
+
+      if exclude_roles == 'exclude_admins_and_moderators'
+        pageviews = pageviews
+          .joins('LEFT JOIN impact_tracking_sessions ON impact_tracking_pageviews.session_id = impact_tracking_sessions.id')
+          .where("impact_tracking_sessions.highest_role IS NULL OR impact_tracking_sessions.highest_role = 'user'")
+      end
+
+      # We first remove the 'en' locale from the list of supported locales,
+      # because we want to make sure it is the last case in the SQL query.
+      # If it is not the last case, it will match other locales that start with 'en',
+      # like 'en-US' or 'en-GB'.
+      locales_without_en = CL2_SUPPORTED_LOCALES.reject { |locale| locale.to_s == 'en' }
+
+      cases = locales_without_en.map do |locale|
+        "WHEN starts_with(path, '/#{locale}') THEN '#{locale}'"
+      end
+
+      # We add the 'en' locale manually as the last case in the SQL query.
+      cases << "WHEN starts_with(path, '/en') THEN 'en'"
+
+      locales = pageviews
+        .select(
+          'count(distinct session_id) as count, ' \
+          "CASE #{cases.join(' ')} ELSE NULL END as locale"
+        )
+        .group(:locale)
+
+      {
+        sessions_per_locale: locales.each_with_object({}) do |row, obj|
+          locale = row['locale']
+
+          if locale.present?
+            count = row['count'].to_i
+            obj[locale] = count
+          end
+        end
+      }
```

Comment:
I tried to simplify it a bit, but didn't test it thoroughly:
```suggestion
      locale_sql = Arel.sql("split_part(path, '/', 2)")
      session_locales = pageviews
        .distinct.select(:session_id, locale_sql.as('locale'))
        .where(locale_sql.in(CL2_SUPPORTED_LOCALES))

      sessions_per_locale = ImpactTracking::Session
        .with(session_locales: session_locales)
        .from('session_locales')
        .group(:locale)
        .count

      { sessions_per_locale: }
```